WoWsShipBuilder.Web/Dialogs/BuildConfigurationDialog.razor

```diff
@@ -303,8 +314,78 @@
         buildNameDictionary[SelectedContainer.Id] = string.IsNullOrEmpty(ViewModel!.BuildName) ? ShipComparisonViewModel.DefaultBuildName : ViewModel.BuildName;
         ViewModel = null;
         StateHasChanged();
-        currentBuildString = string.Empty;
+        inputBuildString = string.Empty;
         selectedTab = newIndex;
         UpdateViewModel();
     }
+
+    private async Task<string?> ValidateBuildString(string buildStr)
```

Comment:
Is this the exact same validation logic as in the build string dialog? If yes, extract logic into a shared method